benchmarks/benchmarks/preprocessing.py

```diff
@@ -7,116 +7,131 @@
 
 from typing import TYPE_CHECKING
 
+from asv_runner.benchmarks.mark import skip_for_params
+
 import scanpy as sc
 from scanpy.preprocessing._utils import _get_mean_var
 
-from .utils import lung93k, pbmc3k, pbmc68k_reduced
+from .utils import get_dataset
 
 if TYPE_CHECKING:
     from anndata import AnnData
 
+    from .utils import Dataset
+
 
 adata: AnnData
+batch_key: str | None
 
 
-def setup(*params: str):
-    """Setup all tests.
+def setup(dataset: Dataset, *_):
+    """Setup global variables before each benchmark."""
+    global adata, batch_key
+    adata, batch_key = get_dataset(dataset)
 
-    The SparseDenseSuite below defines a parameter,
-    the other tests none.
-    """
-    global adata
 
-    if len(params) == 0 or params[0] == "pbmc68k_reduced":
-        adata = pbmc68k_reduced()
-    elif params[0] == "pbmc3k":
-        adata = pbmc3k()
-    elif params[0] == "lung93k":
-        adata = lung93k()
-    else:
-        raise ValueError(f"Unknown dataset {params[0]}")
+# The actual test suite begins here
 
+params: list[Dataset] = ["pbmc68k_reduced", "pbmc3k"]
+param_names = ["dataset"]
```

Comment:
Why only the two?

Reply:
Ideally we want to have useful benchmarks, that cover all code paths we handle separately.

This PR is for the code paths “dense in-memory dataset” and “sparse in-memory dataset” (dask comes later)

I think if we have separate code paths for bigger datasets in some functions, we should benchmark these separately, with a bigger dataset.